pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -359,6 +359,69 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func BenchmarkMoveAllToActiveOrBackoffQueue(b *testing.B) {
+	tests := []struct {
+		name    string
+		podsNum []int
+	}{
+		{
+			name:    "1k-pods",
+			podsNum: []int{200, 800},
+		},
+		{
+			name:    "5k-pods",
+			podsNum: []int{1000, 4000},
+		},
+	}
+
+	podTemplates := []v1.Pod{highPriorityPod, highPriNominatedPod, medPriorityPod, unschedulablePod}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			c := clock.NewFakeClock(time.Now())
+			q := NewPriorityQueue(newDefaultQueueSort(), WithClock(c))
+
+			// Init pods in activeQ.
+			for i := 0; i < tt.podsNum[0]; i++ {
+				p := podTemplates[i%len(podTemplates)].DeepCopy()
+				p.Name, p.UID = fmt.Sprintf("%v-%v", p.Name, i), types.UID(fmt.Sprintf("%v-%v", p.UID, i))
+				q.Add(p)
+			}
+			// Init pods in unschedulableQ.
+			for i := 0; i < tt.podsNum[1]; i++ {
+				p := podTemplates[i%len(podTemplates)].DeepCopy()
+				p.Name, p.UID = fmt.Sprintf("%v-%v", p.Name, i+tt.podsNum[0]), types.UID(fmt.Sprintf("%v-%v", p.UID, i+tt.podsNum[0]))
+				podInfo := q.newQueuedPodInfo(p)
+				podInfo.Attempts = 1
+				q.AddUnschedulableIfNotPresent(podInfo, q.SchedulingCycle())
+			}
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				// Iterate 10 times, each time we step 2 seconds forward, and run flushBackoffQCompleted()
+				// every 5 iterations (10 seconds).
+				for j := 1; j <= 10; j++ {
+					// Moves clock by 2 seconds.
+					c.Step(time.Second * 2)
+					q.MoveAllToActiveOrBackoffQueue("test")
+
+					// Pop half of the pods in activeQ and re-add them.
```

Comment:
```suggestion
					// Pop half of the pods in activeQ and expect to re-add them in the next iteration.
```

Reply:
I added a line below to explain the intention here.